README.md

```diff
@@ -12,167 +12,218 @@ Using Kubernetes primitives, administrators configure identities and bindings to
+
+Here is an example cURL command:
+
+```bash
+curl http://127.0.0.1:2579/host/token/?resource=https://vault.azure.net -H "podname: nginx-flex-kv-int" -H "podns: default"
+```
+
+For more information, please refer to the [design documentation](./docs/design/concept.md).
+
+## Demo
+
+You will need a Kubernetes cluster running on Azure, either managed by [AKS] or provisioned with [AKS Engine].
+
+You will also need to install the following tools:
+- [Azure CLI]
+- [jq]
 
-You will need a Kubernetes cluster running on Azure, either managed by [AKS] or provisioned with [AKS Engine]. 
+> If running MSI-enabled AKS cluster, please make sure to follow the [MSI specific](docs/readmes/README.msi.md) documentation before getting started.
 
-**Note** If running MSI-enabled AKS cluster, please make sure to follow the [MSI specific](docs/readmes/README.msi.md) documentation before getting started.
+Set the following Azure-related environment variables:
 
-### 1. Create the Deployment
+```bash
+export SUBSCRIPTION_ID="<SubscriptionId>"
+export RESOURCE_GROUP="<ResourceGroup>"
+export IDENTITY_NAME="demo"
+```
 
-AAD Pod Identity consists of the Managed Identity Controller (MIC) deployment, the Node Managed Identity (NMI) daemon set, and several standard and custom resources. For more information, see [Components].
+### 1. Deploy aad-pod-identity
 
-Run this command to create the `aad-pod-identity` deployment on an RBAC-enabled cluster:
+Deploy `aad-pod-identity` components to an RBAC-enabled cluster:
 
-```shell
+```bash
 kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment-rbac.yaml
 ```
 
-Or run this command to deploy to a non-RBAC cluster:
+Deploy `aad-pod-identity` components to a non-RBAC cluster:
 
-```shell
+```bash
 kubectl apply -f https://raw.githubusercontent.com/Azure/aad-pod-identity/master/deploy/infra/deployment.yaml
 ```
 
 > Important: For AKS clusters with limited [egress-traffic], Please install pod-identity in `kube-system` namespace using the [helm charts].
 
-### 2. Create an Azure Identity
+### 2. Create an identity on Azure
 
-Run this [Azure CLI] command, and take note of the `clientId` and `id` values it returns:
+Create an identity on Azure and store the client ID and resource ID of the identity as environment variables.
 
-```shell
-az identity create -g <resourcegroup> -n <name> -o json
+```bash
+OUTPUT="$(az identity create --resource-group $RESOURCE_GROUP --name $IDENTITY_NAME -ojson)"
+export IDENTITY_CLIENT_ID="$(jq -r '.clientId' <<< $OUTPUT)"
```

Comment:
can we use the az command to get these instead of jq? It's possible some users might not have it

```
➜ az identity show -n <identity name> -g <resource group> --query clientId -o tsv
➜ az identity show -n <identity-name>-g <resource group> --query id -o tsv
```